front/app/containers/ProjectsShowPage/shared/survey/SurveyXact.tsx

```diff
@@ -44,12 +40,30 @@ type Props = {
 
 export default memo<Props>(({ surveyXactUrl, className }) => {
   const [isIframeLoaded, setIsIframeLoaded] = useState(false);
+  const [hackyWidthThingy, setHackyWidthThingy] = useState<string>('100%');
   const windowSize = useWindowSize();
   const smallerThanLargeTablet = windowSize
     ? windowSize.windowWidth <= viewportWidths.largeTablet
     : false;
+
   const handleIframeOnLoad = () => {
     setIsIframeLoaded(true);
+    setTimeout(() => {
+      setHackyWidthThingy(
+        document.getElementById('survey-xact-frame')?.clientWidth.toString() ||
+          '100%'
+      );
+      setTimeout(() => {
+        setHackyWidthThingy((width) =>
+          width === '100%'
+            ? document
+                .getElementById('survey-xact-frame')
+                ?.clientWidth.toString() || '99%'
+            : (parseInt(width, 10) - 1).toString()
+        );
+      }, 1000);
+    }),
+      10000;
```

Comment:
Not sure if helpful/already aware of this, but in components/Fragment we do the following to set the iframe height:

```
  setIframeRef = (ref) => {
    this.iframeNode = ref;
  };

  setIframeHeight = () => {
    if (this.iframeNode && this.iframeNode.contentWindow) {
      this.setState({
        iframeHeight:
          this.iframeNode.contentWindow.document.body.scrollHeight * 1.1,
      });
    }
  };

  render() {
    const { children, title, className } = this.props;
    const { iframeHeight } = this.state;

    return (
      <StyledIframe
        className={className}
        title={title}
        ref={this.setIframeRef}
        src={this.fragmentUrl()}
        height={iframeHeight}
        onLoad={this.setIframeHeight}
      />
    );
  }
```

Reply:
Indeed, with the first timeout, I'm awaiting for the element to be there and for their js to be ready to react to the change, and with the second one, I'm changing the width. 
Thanks for the pointers ! Will try to fix Iframe height with this idea, we'll see if it works because content-policies sometimes block access to underlying iframes. I'll leave the width hack as it is, already spent enough time on this, and tried a whole lot of counters that sometimes got reset... Hopefully the survey provider will fix and this is temporary anyways.

Reply:
Yes, just tried it out and it's blocked by content policy, something to look into, but not today :)